cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/SourceControlManagementHttpHandlerTests.java

```diff
@@ -180,6 +215,40 @@ public void testFeatureFlagDisabledEndpoint() throws Exception {
     assertResponseCode(403, deleteRepository(NAME));
   }
 
+  @Test
+  public void testValidateRepoConfigSuccess() throws Exception {
+    RepositoryConfig config = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+                                .setLink(LINK)
+                                .setDefaultBranch(DEFAULT_BRANCH)
+                                .setAuthType(AuthType.PAT)
+                                .setTokenName(TOKEN_NAME)
+                                .build();
+
+    Mockito.doNothing()
+      .when(getInjector().getInstance(SourceControlManagementService.class))
+      .validateRepository(Mockito.any(), Mockito.any());
+    
+    HttpResponse response = setRepository(NAME, createRepoValidateString(config));
+    assertResponseCode(200, response);
+  }
+
+  @Test
+  public void testValidateRepoConfigFails() throws Exception {
+    RepositoryConfig config = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+                                .setLink(LINK)
+                                .setDefaultBranch(DEFAULT_BRANCH)
+                                .setAuthType(AuthType.PAT)
+                                .setTokenName(TOKEN_NAME)
+                                .build();
+
+    Mockito.doThrow(new RepositoryConfigValidationException("validation failed"))
```

Comment:
should use an exception with some errors, then check that the response contains the errors.